### PR TITLE
Draw state cpu

### PR DESCRIPTION
**Pull request by davidpiuva**
InterpolationMode is replaced with a boolean flag to keep it simple and reduce code bloat.
The legacy draw method is now private and called from Draw~DrawState in RasterCanvas.
Viewport is used without destination to avoid the risk of floating point rounding errors in exact memory addresses.

Solves https://github.com/magic-lang/ooc-kean/issues/857

### CODE REVIEW

**Comment by davidpiuva** [comment]
@emilwestergren Peer review?

**Comment by marcusnaslund** [comment]
You seem to be getting conflicts in a corresponding branch (do you have one with the same name there?) in the other project.

**Comment by davidpiuva** [comment]
Both has the same branch name and both depends on each other.

**Comment by davidpiuva** [comment]
Fixed the failing tests.

**Comment by davidpiuva** [comment]
@marcusnaslund All checks passed.

File: source/draw/RasterMonochrome.ooc [hunk 1]
```diff
@@ -68,14 +71,11 @@ RasterMonochrome: class extends RasterPacked {
...
 	resizeTo: override func (size: IntVector2D) -> This {
-		this resizeTo(size, InterpolationMode Smooth) as This
+		this resizeTo(size, true) as This
 	}
-	resizeTo: override func ~withMethod (size: IntVector2D, method: InterpolationMode) -> This {
+	resizeTo: override func ~withMethod (size: IntVector2D, interpolate: Bool) -> This {
 		result := This new(size)
-		result canvas interpolationMode = method
-		result canvas draw(this, IntBox2D new(this size), IntBox2D new(size))
-		//TODO will be fixed by subcanvas
```
Comment (davidpiuva) [inline source/draw/RasterMonochrome.ooc]: Was fixed by DrawState.

File: test/draw/RasterCanvasTest.ooc [hunk 2]
```diff
@@ -109,15 +109,13 @@ RasterCanvasTest: class extends Fixture {
 			output := "test/draw/output/RasterCanvas_drawYUVonRGB.png"
 			imageFlower := RasterYuv420Semiplanar open(this inputFlower)
 			outputImage := RasterRgb open(inputSpace)
-			outputImage canvas interpolationMode = InterpolationMode Smooth
-			outputImage canvas draw(imageFlower, IntBox2D new(imageFlower size), IntBox2D new(20, 30, 100, 250))
+			DrawState new(outputImage) setInputImage(imageFlower) setViewport(IntBox2D new(20, 30, 100, 250)) setInterpolate(true) draw()
 			imageFlowerYUpward := RasterYuv420Semiplanar open(this inputFlower, CoordinateSystem YUpward)
-			outputImage canvas draw(imageFlowerYUpward, IntBox2D new(imageFlower size), IntBox2D new(130, 30, 100, 250))
-			outputImage canvas interpolationMode = InterpolationMode Fast
+			DrawState new(outputImage) setInputImage(imageFlowerYUpward) setViewport(IntBox2D new(130, 30, 100, 250)) setInterpolate(true) draw()
```
Comment (marcusnaslund) [inline test/draw/RasterCanvasTest.ooc]: Shouldn't it be `setInterpolate(false)`?

**Comment by marcusnaslund** [comment]
Looks good.

**Pull request merged**